cmd/cpu-device-plugin/cpu-device-plugin.go

```diff
@@ -206,20 +206,29 @@ func createPluginsForPools() error {
 			sharedCPUs = pool.CPUs.String()
 		}
 		cdm := newCPUDeviceManager(poolName, pool, sharedCPUs)
-		if err := cdm.Start(); err != nil {
-			glog.Errorf("cpuDeviceManager.Start() failed: %v", err)
-			break
-		}
-		resourceName := resourceBaseName + "/" + poolName
-		err := cdm.Register(path.Join(pluginapi.DevicePluginPath, "kubelet.sock"), resourceName)
-		if err != nil {
-			// Stop server
-			cdm.grpcServer.Stop()
-			glog.Error(err)
-			break
-		}
 		cdms = append(cdms, cdm)
-		glog.Infof("CPU device plugin registered with the Kubelet")
+	}
+	if err != nil {
+		return err
+	}
+	for poolName, pool := range poolConf.Pools {
+		for _, cdm := range cdms{
+			if pool.CPUs.Equals(cdm.pool.CPUs){
```

Comment:
why is this extra condition needed? I don't see it in the original code

Reply:
To make sure that cdms will register with the corresponding resource name

Reply:
yeah but if you also don't even create te CDMs in the first loop, you don't need to iterate over the CDMs for every pool in the second
if you anyway need to iterate over the Pools again, it doesn't make any sense

moving CDM creation and append to the second loop decreases its complexity from O(nm) to O(n)